### PR TITLE
Performance fixes relating to floating point operations

### PR DESCRIPTION
**Pull request by nsajko**
This replaces unnecessary double-precision soft-float operations with
single-precision floating-point operations, mainly by casting.

In a couple places I also replace a function call with a constant.

### CODE REVIEW

**Comment by nsajko** [comment]
There are some (two I think) cases involving overloaded functions where double precision is used for a reason I can not grasp yet. (I guess I am not that good with C++.) But let that be a future issue.

**Comment by nsajko** [comment]
Can I get help with the failing test? The iOS build is failing, but I am not being shown why...

**Comment by nsajko** [comment]
An accuracy test is failing on macos, but that probably just means the test should be fixed. The particular example in the test seems OK accuracy-wise, and indeed all the changes in this PR should be, as they basically just change the code to work as intended (use floats when supposed to, instead of using doubles just for some intermediate calculations).

Note that I have nothing against using doubles when necessary for greater accuracy, but at least most of the changes here are obviously a good trade-off. For the future: it would probably be a good idea to think of a way to make using soft-float double arithmetic more explicit.

**Comment by nsajko** [comment]
Also, it would be good if you Numworks people could point me to some kind of document that explained your expectations regarding when to use double, and when to use float. Maybe you could make a whitelist of all the places where double may be used?

**Comment by nsajko** [comment]
@LeaNumworks 

Turtle::drawPaw (python/port/mod/turtle/turtle.cpp) uses `float`s, and this seems incongruent with the fact that Epsilon's MicroPython is configured to use `double`s. Should I change one or the other?

File: liba/src/external/openbsd/s_logbf.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/* s_logbf.c -- float version of s_logb.c.
```
Comment (LeaNumworks) [inline liba/src/external/openbsd/s_logbf.c]: Hello,
We prefer to use OpenBSD 4.9's libm version (see 3adc5d9fac733fe04a4), and also not modifying the external files. Could you please replace this file with [s_logbf.c.zip](https://github.com/numworks/epsilon/files/4254640/s_logbf.c.zip)?

File: liba/src/external/openbsd/s_erff.c [hunk 2]
```diff
@@ -0,0 +1,185 @@
+/* s_erff.c -- float version of s_erf.c.
```
Comment (LeaNumworks) [inline liba/src/external/openbsd/s_erff.c]: Same comment as for s_logbf.c, could you please change this file with [s_erff.c.zip](https://github.com/numworks/epsilon/files/4254712/s_erff.c.zip)?

**Comment by LeaNumworks** [comment]
> @LeaNumworks
> 
> Turtle::drawPaw (python/port/mod/turtle/turtle.cpp) uses `float`s, and this seems incongruent with the fact that Epsilon's MicroPython is configured to use `double`s. Should I change one or the other?

I see your point! However, I do not think it is necessary to change to `double`, as we are doing computations that should be fast and do not need precision.

**Comment by LeaNumworks** [comment]
> An accuracy test is failing on macos, but that probably just means the test should be fixed. The particular example in the test seems OK accuracy-wise, and indeed all the changes in this PR should be, as they basically just change the code to work as intended (use floats when supposed to, instead of using doubles just for some intermediate calculations).
> 
> Note that I have nothing against using doubles when necessary for greater accuracy, but at least most of the changes here are obviously a good trade-off. For the future: it would probably be a good idea to think of a way to make using soft-float double arithmetic more explicit.

I pushed a commit to change the precision of this test, it should pass now. (While at it, I fixed the precision of some surrounding tests too).

File: liba/src/external/openbsd/s_logbf.c (hunk 1 above)
Comment (nsajko) [reply liba/src/external/openbsd/s_logbf.c]: Sadly, that logbf (and logb) is wrong for subnormal inputs (it returns a constant). Could you import just this file from openlibm?

I started a discussion on the OpenBSD misc mailing list regarding this bug: https://marc.info/?l=openbsd-misc&m=158272385126733&w=2

**Comment by LeaNumworks** [comment]
Hi @nsajko, sorry for the delay!
As stated in #1370, we've decided we'd rather use OpenBSD. I created PR #1486, which rebases this one and uses the OpenBSD files.
I will close this PR and merge the rebased one.
Feel free to open issues if you see any bug resulting from the OpenBSD faulty files, and thank you for emailing OpenBSD about the problem!

**Pull request closed without merging**